kbase-extension/static/kbase/js/common/cellComponents/tabs/results/reportWidget.js

```diff
@@ -31,39 +31,61 @@ define(['bluebird', 'jquery', 'common/html', 'common/ui', 'common/events', 'kbas
          */
         function renderReports(objectData, events) {
             return objectData.map((objInfo) => {
-                return div([
-                    a(
+                // clean up objInfo before rendering / passing to toggleReportView
+                let name = objInfo.name;
+                if (!name) {
+                    if (objInfo.ref) {
+                        name = `Object ${objInfo.ref} not found, may have been deleted`;
+                    } else {
+                        name = 'Object not found';
+                    }
+                }
+
+                let reportElem = '';
+                if (objInfo.reportRef) {
+                    reportElem = a(
                         {
                             class: 'kb-report__toggle collapsed',
                             dataToggle: 'collapse',
                             ariaExpanded: false,
                             id: events.addEvent({
                                 type: 'click',
-                                handler: (e) => toggleReportView(e, objInfo),
+                                handler: (e) => toggleReportView(e, objInfo.reportRef),
                             }),
                         },
-                        objInfo.name
-                    ),
-                ]);
+                        name
+                    );
+                } else {
+                    reportElem = name + ' (report not found)';
```

Comment:
again, could init reportElem with this value and change it if `objInfo.reportRef` is available